gateway/ln-gateway/src/rpc/lnrpc_client.rs

```diff
@@ -142,20 +144,81 @@ impl ILnRpcClient for NetworkLnRpcClient {
     async fn subscribe_intercept_htlcs(
         &self,
         short_channel_id: u64,
-    ) -> Result<mpsc::Receiver<HtlcInterceptPayload>> {
+    ) -> Result<mpsc::Receiver<GatewayChannelMessage>> {
         let mut client = self.client.lock().await.clone();
-
-        let mut response_stream = client
+        let mut stream = client
             .subscribe_intercept_htlcs(Request::new(SubscribeInterceptHtlcsRequest {
                 short_channel_id,
             }))
             .await
             .expect("Failed to subscribe intercept htlcs")
             .into_inner();
 
-        let (_, rx) = mpsc::channel::<HtlcInterceptPayload>(100);
-
-        // TODO: Send intercepted HTLCs from stream to gateway actors for processing
+        // Spawn a task to listen for messages from the htlc stream.
+        // Adapt HTLC data from protobuf types to gateway types and send for processing.
+        let mut tg = self.task_group.clone();
+        let (tx, rx) = mpsc::channel::<GatewayChannelMessage>(100);
+
+        tg.spawn(
+            "Subscribe to HTLC intercept stream",
+            move |subscription| async move {
+                loop {
+                    // Shutdown the task group when the stream is closed
+                    if subscription.is_shutting_down() {
+                        break;
+                    }
+
+                    let mut htlc_outcomes = Vec::<CompleteHtlcsRequest>::new();
```

Comment:
This vector seems to keep growing indefinitely, which will crash the gateway eventually. I also can't find its use. This is related to if the while loop ever finishes.